AzureFirewall/AzureFirewallApplicationRule.kql

```diff
@@ -0,0 +1,38 @@
+let applicationRuleAllowsHttp = AzureDiagnostics
+| where ResourceType == "AZUREFIREWALLS"
+| where Category == "AzureFirewallApplicationRule"
+| parse msg_s with * " request from " ip_source ":" port_source " to " ip_destination ":" port_destination ". Url: " url ". Action: " action ". Policy: " policy ". Rule Collection Group: " rulecollectiongroup "Rule Collection: " rulecollection ". Rule: " rule
+| where action == "Allow"
+| project-away msg_s;
+let applicationRuleAllowsHttps = AzureDiagnostics
+| where ResourceType == "AZUREFIREWALLS"
+| where Category == "AzureFirewallApplicationRule"
+| parse msg_s with * " request from " ip_source ":" port_source " to " ip_destination ":" port_destination ". Action: " action ". Policy: " policy ". Rule Collection Group: " rulecollectiongroup "Rule Collection: " rulecollection ". Rule: " rule
+| where action == "Allow" and port_destination !hasprefix "80" and rule !has "Web Category"
+| project-away msg_s;
+let applicationRuleAllowsHttpsWebCategory = AzureDiagnostics
+| where ResourceType == "AZUREFIREWALLS"
+| where Category == "AzureFirewallApplicationRule"
+| parse msg_s with * " request from " ip_source ":" port_source " to " ip_destination ":" port_destination ". Action: " action ". Policy: " policy ". Rule Collection Group: " rulecollectiongroup "Rule Collection: " rulecollection ". Rule: " rule ". Web Category: " webcategory
+| where action == "Allow"
+| project-away msg_s;
+let applicationRuleDeniesHttp = AzureDiagnostics
+| where ResourceType == "AZUREFIREWALLS"
+| where Category == "AzureFirewallApplicationRule"
+| parse msg_s with * " request from " ip_source ":" port_source " to " ip_destination ":" port_destination ". Url: " url ". Action: " action ". " rule ". " *
+| where action == "Deny" and port_destination == 80
+| project-away msg_s;
+let applicationRuleDeniesHttps = AzureDiagnostics
+| where ResourceType == "AZUREFIREWALLS"
+| where Category == "AzureFirewallApplicationRule"
+| parse msg_s with * " request from " ip_source ":" port_source " to " ip_destination ":" port_destination ". Action: " action ". " rule ". " *
+| where action == "Deny" and port_destination !hasprefix "80"
+| project-away msg_s;
+let applicationRuleDenies = AzureDiagnostics
+| where ResourceType == "AZUREFIREWALLS"
+| where Category == "AzureFirewallApplicationRule"
+| parse msg_s with * " request from " ip_source ":" port_source ". Action: " action ". Reason: " rule "." *
+| where isnotempty(ip_source)
+| project-away msg_s;
+applicationRuleAllowsHttp
+| union applicationRuleAllowsHttps, applicationRuleAllowsHttpsWebCategory, applicationRuleDeniesHttp, applicationRuleDeniesHttps, applicationRuleDenies
```

Comment:
It would make sense to separate proxy (with URL) and network events (without URL) into different parsers as commonly queries need either but not both (and can union in the query if otherwise).

Reply:
Yes, you're right. That's what I did now with the new version.